be/src/util/block_compression.cpp

```diff
@@ -906,8 +919,13 @@ class ZlibBlockCompression : public BlockCompressionCodec {
         size_t input_size = input.size;
         auto zres =
                 ::uncompress2((Bytef*)output->data, &output->size, (Bytef*)input.data, &input_size);
-        if (zres != Z_OK) {
+        if (zres == Z_DATA_ERROR) {
             return Status::InvalidArgument("Fail to do ZLib decompress, error={}", zError(zres));
+        } else if (zres == Z_MEM_ERROR) {
+            return Status::MemoryLimitExceeded("Fail to do ZLib decompress, error={}",
+                                               zError(zres));
+        } else if (zres == Z_BUF_ERROR) {
+            return Status::InternalError("Fail to do ZLib decompress, error={}", zError(zres));
         }
```

Comment:
I think here should be a cover branch of zres != Z_OK. and check other changes for this problem